scala-ide-scaps-plugin/src/main/scala/scaps/eclipse/core/services/ScapsService.scala

```diff
@@ -24,15 +24,17 @@ import org.eclipse.core.resources.ResourcesPlugin
 
 object ScapsService {
 
-  val pluginPreferences = InstanceScope.INSTANCE.getNode(ScapsPlugin.PLUGIN_ID)
-  val indexRootDir = ResourcesPlugin.getWorkspace.getRoot.getLocation.toString + ScapsPlugin.INDEX_RELATIVE_ROOT_DIR
+  private val pluginPreferences = InstanceScope.INSTANCE.getNode(ScapsPlugin.PLUGIN_ID)
+  private val indexRootDir = ResourcesPlugin.getWorkspace.getRoot.getLocation.toString + ScapsPlugin.INDEX_RELATIVE_ROOT_DIR
 
   val PROPERTY_INDEXER_RUNNING = "indexRunning"
 
   val PROPERTY_SEARCH_ON_FIRST_INDEX = "searchOnFirstIndex"
   val FIRST_INDEX_DIR = "first"
   val SECOND_INDEX_DIR = "second"
 
+  // Könnte es hier zu Threading-Problemen kommen? Race condition, etc.?
+
```

Comment:
Die Primitiven die Scala bietet sind gleich wie bei Java; Semaphor wäre möglich, aber auch synchronized (das ist allerdings kein Keyword in Scala, sondern einfach eine Methode, also this.synchronized { ... }).